src/node_buffer.cc

```diff
@@ -818,19 +813,16 @@ void WriteFloatGeneric(const FunctionCallbackInfo<Value>& args) {
   // throw an error (if needed) and return directly.
   if (!ParseArrayIndex(args[2], 0, &offset, memcpy_num) ||
       offset >= ts_obj_length) {
-    if (should_assert)
-      THROW_AND_RETURN_IF_OOB(false);
+    THROW_AND_RETURN_IF_OOB(false);
     return;
   }
 
   // If the offset is too large for the entire value, but small enough to fit
-  // part of the value, throw an error and return only if should_assert is
-  // true. Otherwise, write the part of the value that fits.
+  // part of the value, throw an error and return.  Otherwise, write the part
+  // of the value that fits.
   if (offset + memcpy_num > ts_obj_length) {
-    if (should_assert)
-      THROW_AND_RETURN_IF_OOB(false);
-    else
-      memcpy_num = ts_obj_length - offset;
+    THROW_AND_RETURN_IF_OOB(false);
+    memcpy_num = ts_obj_length - offset;
   }
```

Comment:
Uhh, the `memcpy_num` assignment is dead code since `THROW_AND_RETURN_IF_OOB` will always execute because of `false`.